### ISSUE TITLE
Support using LLM model via Groq cloud

### ISSUE THREAD

**Issue by PeterDaveHello** [vscode-reborn-ai/vscode-reborn-ai#116]
### Describe the feature

It'd be awesome if we could add support for Groq Cloud. It's fast and has some excellent new models, including Llama 3.1 405B(https://console.groq.com/docs/models#llama-31-405b-preview), which is even currently free.

The good news is that Vercel AI SDK, which we've been switched to, already supports Groq(https://sdk.vercel.ai/providers/ai-sdk-providers/groq), so implementing this shouldn't be too tricky.

We could integrate Groq Cloud API and add options for users to choose from Groq's models. This would give our users more choices and access to more powerful models.

What do you guys think?

**Comment by Christopher-Hayes** [comment]
Sounds good to me. I do want to support as many services/APIs as possible.

I was really confused at first, thinking this was related to Twitter's Grok, lol.

@PeterDaveHello If you want to work on this one, feel free to assign yourself to this issue. If you don't have the access to do that, lmk.

**Comment by PeterDaveHello** [comment]
Great! I'll assign myself when I have some spare time; before that, anyone can volunteer to do it!

**Comment by PeterDaveHello** [comment]
> Sounds good to me. I do want to support as many services/APIs as possible.

By the way, Groq doesn't provide the most LLM models; they are fast and free right now. So, to support as many as possible, here are some other service providers you may want to take a look:

- https://www.together.ai/
- https://fireworks.ai/
- https://openrouter.ai/

**Comment by Christopher-Hayes** [comment]
> > Sounds good to me. I do want to support as many services/APIs as possible.
> 
> By the way, Groq doesn't provide the most LLM models; they are fast and free right now. So, to support as many as possible, here are some other service providers you may want to take a look:
> 
>     * https://www.together.ai/
> 
>     * https://fireworks.ai/
> 
>     * https://openrouter.ai/

Yeah, Together AI and Fireworks I'd like to support. We already support OpenRouter, it works pretty great. If you use OpenRouter, you might already be using fireworks or together because OpenRouter just routes you to those providers for most models. OpenRouter self-hosts a small subset of models.

In another PR I'm working on support for Featherless, they self-host a TON of models. They use cloud GPUs like everyone else, but they swap out models in realtime, which allows them support 1,700+ models.

**Comment by PeterDaveHello** [comment]
Another attractive model that does not use a transformer-based model that you may want to take a look at is Codestral Mamba: https://mistral.ai/news/codestral-mamba/, but it's not so wildly deployed yet.

**Comment by Christopher-Hayes** [comment]
The `ai` package does support mistral, so that could be another one.
https://sdk.vercel.ai/docs/foundations/providers-and-models

Honestly, I'm thinking we should find a way in the UI to represent the "best" models users can use. For example, a list somewhere with like "Use Mistral's 'Codestral Mambra' with either Mistral, OpenRouter, or ... for some of the best code refactoring". I guess that's a future thing we can work on, because I expect users may not know what LLMs are best for their need.

**Comment by PeterDaveHello** [comment]
For most users, GPT-4o, Claude-3.5-Sonnet & Gemini-1.5-Pro are those best they can have 😄

**Comment by PeterDaveHello** [comment]
> The `ai` package does support mistral, so that could be another one. [sdk.vercel.ai/docs/foundations/providers-and-models](https://sdk.vercel.ai/docs/foundations/providers-and-models)

Though it might be a little bit too late, maybe langchainjs can give us the wildest support?

See:
- https://github.com/langchain-ai/langchainjs/tree/main/libs/
- https://github.com/langchain-ai/langchainjs/tree/main/libs/langchain-mistralai
